onnxruntime/core/providers/webgpu/nn/pool.cc

```diff
@@ -0,0 +1,254 @@
+    if (!count_include_pad_) {
+      sampling_ss << "      count++;\n";
+    }
+    sampling_code = sampling_ss.str();
+
+    std::stringstream downsampling_ss;
+    downsampling_ss << "  value /= " << (is_float16_ ? "f16" : "f32") << "(count);\n";
+    downsampling_code = downsampling_ss.str();
+  }
+
+  const auto kernel_rank = kernel_shape_.size();
+  const auto pads_rank = kernel_shape_.size() * 2;
+  // The dimension index for H or D1
+  const auto data_dim_begin = is_nhwc_ ? 1 : 2;
+  // The dimension index after W or Dn
+  auto data_dim_end = input.Rank();
+  data_dim_end = is_nhwc_ ? data_dim_end - 1 : data_dim_end;
+
+  std::stringstream d_idx_ss;
+  d_idx_ss << "j - " << data_dim_begin;
+  std::string d_idx_code = d_idx_ss.str();
+
+  // clang-format off
+  auto& body = shader.MainFunctionBody();
+  body << shader.GuardAgainstOutOfBoundsWorkgroupSizes("uniforms.output_size")
+       << "  let y_indices = " << output.OffsetToIndices("global_idx") << ";\n"
+       << "  var x_indices = y_indices;\n"
+       << "  var k_indices: array<u32, " << kernel_rank << ">;\n"
+       <<    var_decl_code
+       << "  for (var i: u32 = 0; i < uniforms.kernel_size; i++) {\n"
+       << "    var offset = i;\n"
+       // ---- Compute offset to indices in pooling window.
+       << "    for (var j = 0; j < " << kernel_rank << "; j++) {\n"
+       << "      k_indices[j] = offset / " << GetElementAt("uniforms.kernel_strides", "j", kernel_rank) << ";\n"
+       << "      offset = offset % " << GetElementAt("uniforms.kernel_strides", "j", kernel_rank) << ";\n"
+       << "    }\n"
+       // ---- Apply dilations in pooling window.
+       << "    for (var j = 0; j < " << kernel_rank << "; j++) {\n"
+       << "      k_indices[j] *= " << GetElementAt("uniforms.dilations", "j", kernel_rank) << ";\n"
+       << "    }\n"
+       << "    var is_pad = false;\n"
+       // ---- Compute x_indices in each data dimension
+       << "    for (var j = " << data_dim_begin << "; j < " << data_dim_end << "; j++) {\n"
+       << "      x_indices[j] = y_indices[j] * " << GetElementAt("uniforms.strides", d_idx_code, kernel_rank) << ";\n"
+       << "      x_indices[j] += k_indices[" << d_idx_code << "];\n"
+       << "      x_indices[j] -= " << GetElementAt("uniforms.pads", d_idx_code, pads_rank) << ";\n"
+       << "      let j_dim_len = " << input.IndicesGet("uniforms.input_shape", "j") << ";\n"
+       // ------ Check if x_indices[j] is out of bounds to handle padding.
+       << "      if (x_indices[j] < 0 || x_indices[j] >= j_dim_len) {\n"
+       << "        is_pad = true;\n"
+       << "        break;\n"
+       << "      }\n"
+       << "    }\n"
+       << "    if (!is_pad) {\n"
+       << "      let x_val = " << input.GetByIndices("x_indices") << ";\n"
+       <<        sampling_code
+       << "    }\n"
+       << "  }\n"
+       <<    downsampling_code
+       << "  " << output.SetByOffset("global_idx", "value") << ";\n";
+  // clang-format on
```

Comment:
I understand why disable the formatter. code indent is very important for making it much easier to read.

however, introducing skipping formatter may cause potential extra future maintenance cost and possible debates on different preferences of styles. I want to avoid those things so relying on a formatter is a solution of balance.

I actually tried checkout this PR and let clang-format to perform the format. It's not bad and still good to read. and also if you like you can use the following trick:

<img width="420" alt="{D6B7F86B-939D-4FE4-ACFF-E5256F023B2A}" src="https://github.com/user-attachments/assets/6bd5b08a-cf07-4342-8b6a-b17a25016acd" />

Reply:
Done, fine to me.